### ISSUE TITLE
[BUG] Finite difference gradient transform is zero with Torch interface

### ISSUE THREAD

**Issue by trbromley** [PennyLaneAI/pennylane#4882]
### Expected behavior

Finite difference transform calculates gradient correctly

### Actual behavior

Finite difference transform returns zero gradients

### Additional information

- If `diff_method` is `"finite-diff"`, the gradient is returned correctly
- If `x` has a `dtype=torch.float32`, then `framework_output` has the expected `float32` output, while `transform_output` is `float64`

### Source code

```shell
import pennylane as qml
import torch

dev = qml.device("default.qubit")
x = torch.tensor(0.543, requires_grad=True, dtype=torch.float64)

diff_method = "best"
# diff_method = "finite-diff"  # this works

@qml.qnode(dev, diff_method=diff_method)
def circuit(x):
    qml.RX(x, wires=[0])
    qml.RY(x, wires=[0])
    qml.CNOT(wires=[0, 1])
    return qml.probs(wires=0)

transform_output = qml.gradients.finite_diff(circuit)(x)
framework_output = torch.autograd.functional.jacobian(circuit, x)

torch.allclose(transform_output, framework_output)
```

### System information

```shell
This happens on dev PennyLane
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by trbromley** [comment]
Came up while investigating #2296

**Comment by josh146** [comment]
Closing as this now seems fixed :)
